docs/source/error_code_list.rst

```diff
@@ -1149,6 +1149,32 @@ types you expect.
 
 See :ref:`overloading <function-overloading>` for more explanation.
 
+
+.. _code-overload-cannot-match:
+
+Check error code of overload function signature match
+--------------------------------------------------------------------------
+
+In the case of an overloaded function, if one of the signatures is never accessible, this error may occur.
+An example where this can occur is with the utilization of floats and int types, due to the behavior of integers matching floats in mypy.
```

Comment:
An example involving float/int isn't great because it relies on another subtle part of the type system. Better to give an example that is more basic, such as having the first overload take `object` and the second `int`.

Reply:
Valid point. I made some edits to this- please let me know if that looks ok.